Framework/API/src/Run.cpp

```diff
@@ -42,6 +42,16 @@ Kernel::Logger g_log("Run");
 // Public member functions
 //----------------------------------------------------------------------
 
+boost::shared_ptr<Run> Run::clone() {
```

Comment:
Could this method be `const`? Just in case it is not intentional/constrained by something else.